api/models.py

```diff
@@ -86,3 +87,104 @@ def save(self, *args, **kwargs):
 
     def __str__(self):
         return '{}{}'.format(self.item_code, self.serial_number)
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, name, cohort, slack_handle,
+                    is_active=True, is_admin=False, is_staff=False,
+                    phone_number=None, picture=None, password=None):
+        if not email:
+            raise ValueError("Email address is required")
+        elif not name:
+            raise ValueError("Name is required")
+        elif not cohort:
+            raise ValueError("Cohort is required")
+        elif not slack_handle:
+            raise ValueError("Slack handle is required")
+
+        user = self.model(
+            email=self.normalize_email(email),
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+        )
+
+        user.set_password(password)
+        user.picture = picture
+        user.staff = is_staff
+        user.admin = is_admin
+        user.active = is_active
+        user.phone_number = phone_number
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email, name, cohort,
+                         slack_handle, password=None):
+        user = self.create_user(
+            email=email,
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+            password=password,
+            is_staff=True,
+        )
+        return user
+
+    def create_superuser(self, email, name, cohort,
+                         slack_handle, password=None):
+        user = self.create_user(
+            email=email,
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+            password=password,
+            is_staff=True,
+            is_admin=True
+        )
+        return user
+
+
+class User(AbstractBaseUser):
```

Comment:
any reason you chose to go with `AbstractBaseUser ` instead of `AbstractUser `?

Reply:
@egichuri  `AbstractUser` does not allow direct assignment of the `email` as a `USERNAME_FIELD`, It also has a default `username` field which we don't need as the `USERNAME_FIELD` in our context. 

`AbstractBaseUser` offers more flexibility by allowing creation of a custom `Manager` which may be helpful in the long run.

Reply:
if you use `AbstractUser ` the field you assign to `USERNAME_FIELD ` should override whatever is set on `AbstractUser`. Then you can set the `username` field to `None` on the custom user model. That way you don't need to have custom forms as well. Also for now the only changes i've seen on the custom `Manager` is removal of the username field.
something like 
```
class User(AbstractUser):
    username = None
    email = models.EmailField(_('email address'), unique=True)

    USERNAME_FIELD = 'email'
```
Lemme know if you wanna pair on this.

Reply:
@egichuri I would love to pair with you on this.